lib/rucio/core/did_meta_plugins/elasticsearch_meta.py

```diff
@@ -0,0 +1,406 @@
+
+    def set_metadata_bulk(
+        self,
+        scope: "InternalScope",
+        name: str,
+        meta: dict[str, Any],
+        recursive: bool = False,
+        *,
+        session: "Optional[Session]" = None
+    ) -> None:
+        """
+        Bulk set metadata keys.
+
+        :param scope: the scope of did
+        :param name: the name of the did
+        :param meta: dictionary of metadata keypairs to be added
+        :param recursive: recurse into DIDs (not supported)
+        :param session: The database session in use
+        :raises DataIdentifierNotFound: If the DID is not found.
+        :raises RucioException: If an error occurs while setting the metadata.
+        """
+        doc_id = f"{scope.internal}{name}"
+        try:
+            # Try to get existing metadata
+            existing_meta = self.get_metadata(scope, name)
+        except exception.DataIdentifierNotFound:
+            existing_meta = {
+                'scope': str(scope.external),
+                'name': str(name),
+                'vo': str(scope.vo)
+            }
+        for key, value in meta.items():
+            if key not in IMMUTABLE_KEYS:
+                existing_meta[key] = value
+
+        try:
+            self.client.index(index=self.index, body=existing_meta, id=doc_id, refresh="true")
+        except Exception as err:
+            raise exception.RucioException(err)
+
+    def delete_metadata(
+        self,
+        scope: "InternalScope",
+        name: str,
+        key: str,
+        *,
+        session: "Optional[Session]" = None
+    ) -> None:
+        """
+        Delete a key from metadata.
+
+        :param scope: the scope of did
+        :param name: the name of the did
+        :param key: the key to be deleted
+        :raises DataIdentifierNotFound: If the DID is not found.
+        :raises RucioException: If an error occurs while setting the metadata.
+        """
+        doc_id = f"{scope.internal}{name}"
+        try:
+            # First, get the current document
+            doc = self.client.get(index=self.index, id=doc_id)
+
+            # Check if the key exists in the document
+            if key in doc['_source']:
+                # Use script to remove the field
+                script = {
+                    "script": {
+                        "source": f"ctx._source.remove('{key}')",
+                        "lang": "painless"
+                    }
+                }
+                self.client.update(index=self.index, id=doc_id, body=script)
+        except elastic_exceptions.NotFoundError as err:
+            raise exception.DataIdentifierNotFound(f"No metadata found for DID '{scope}:{name}' not found") from err
+        except Exception as err:
+            raise exception.RucioException(err)
+
+    def list_dids(
+        self,
+        scope: "InternalScope",
+        filters: Union[list[dict[str, Any]], dict[str, Any]],
+        did_type: str = 'collection',
```

Comment:
This could be changed to this to be consistent with the `DIDClient`:

```suggestion
        did_type: Literal['all', 'collection', 'dataset', 'container', 'file'] = 'collection',
```

Ideally we should refactor `Literal['all', 'collection', 'dataset', 'container', 'file']` into `common/types` (maybe calling it `DIDTypesLiteral`) and accessing it from both here and `DIDClient`